pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -1152,6 +1152,7 @@ func doNetworkInstanceActivate(ctx *zedrouterContext,
 			status.Type)
 		err = errors.New(errStr)
 	}
+	status.PrgdUplinkIntf = status.CurrentUplinkIntf
```

Comment:
Isn't CurrentUplinkIntf empty here? It is only set by the probe code AFAICT and you haven't done any probe yet?

Do you want to capture the initial IfNameList[0], or the actual string which was configured (latter could be "uplink")? I don't quite understand whether you are trying to capture cases other than the specifically bound to "ethN" in the config.

Reply:
No, CurrentUplinkIntf will not be empty here. We pick an interface even before probe testing that has atleast one usable IP address. It is done as part of checkNIprobeUplink. Later if probing determines that the picked interface does not work, we change to a working interface.